src/test/java/com/iexec/worker/docker/DockerServiceTests.java

```diff
@@ -176,7 +176,7 @@ void shouldGetUnauthenticatedClientWhenAuthFailureWithCustomRegistry() throws Ex
                 .build();
         when(dockerRegistryConfiguration.getRegistryCredentials(registry))
                 .thenReturn(Optional.of(credentials));
-        doThrow(Exception.class)
+        doThrow(RuntimeException.class)
```

Comment:
Throwing an `IllegalArgumentException` would make sense, don't you think?

Reply:
comment added with the reason behind this necessity